src/sys/socket/mod.rs

```diff
@@ -430,6 +431,12 @@ macro_rules! cmsg_space {
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct RecvErr {
+    pub ee: sock_extended_err,
+    pub offender: sockaddr_in,
```

Comment:
N.B., the similar `IPV6_RECVERR` option produces `sockaddr_in6`es on IPv6 sockets.  Regardless of which patch we take for this, it should probably have IPv6 <-> v4 parity.